### PR TITLE
Cognito auth 5/7 - set username

### PR DESCRIPTION
**Pull request by indiv0**
### Pull Request Description

5th PR for IDE/Cloud authorization with cognito. This PR introduces user username templates + flows + backend wrappers for  setting username.
Forgot Password flows are to be added in next PRs to keep the changes reviewable.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]

File: app/ide-desktop/lib/content/src/dashboard.css [hunk 1]
```diff
@@ -1 +1,675 @@
...
\ No newline at end of file
+/*! tailwindcss v3.2.7 | MIT License | https://tailwindcss.com*/
```
Comment (wdanilo) [inline app/ide-desktop/lib/content/src/dashboard.css]: I was writing about it in other PR review. Please apply other PR reviews and ping me, I'm skipping reviewing this PR for now.

Comment (indiv0) [reply app/ide-desktop/lib/content/src/dashboard.css]: File removed.

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/setUsername.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                <div className="inline-flex items-center justify-center absolute left-0 top-0 h-full w-10 text-gray-400">
+                  <icons.Svg data={icons.PATHS.at} />
+                </div>
+
+                <input
+                  {...bindUsername}
+                  id="username"
+                  type="text"
+                  name="username"
+                  className="text-sm sm:text-base placeholder-gray-500 pl-10 pr-4 rounded-lg border border-gray-400 w-full py-2 focus:outline-none focus:border-blue-400"
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/setUsername.tsx]: lines > 100 chars

File: app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx [hunk 3]
```diff
@@ -232,9 +238,29 @@ export function AuthProvider(props: AuthProviderProps) {
...
+    username: string,
+    email: string
+  ) => {
+    const body: backendService.SetUsernameRequestBody = {
+      userName: username,
+      userEmail: email,
+    };
+
+    /** TODO [NP]: https://github.com/enso-org/cloud-v2/issues/343
+     * Don't create a new API client here, reuse the one from the context. */
```
Comment (wdanilo) [inline app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/auth.tsx]: A TODO/FIXME in general should describe a problem, potential issues derived from that problem, potential solution and shouldd link to issue. Here we have solution and link, but there is no info about what problems the current solution causes. It is performance related?

**Review by wdanilo — APPROVED** [review]

**Review by PabloBuchu — APPROVED** [review]

**Pull request merged**